Bottom: 5bd81bc351caaeb882c5f12d74ea1de314ee315f
Top:    dfd947d3af54ca9d53239a891514910be5653b6b
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-05-19 14:53:35 +0800

Refresh of introduce-skb_ring

---

diff --git a/include/linux/skb_ring.h b/include/linux/skb_ring.h
index 9666451..9d5f8ac 100644
--- a/include/linux/skb_ring.h
+++ b/include/linux/skb_ring.h
@@ -14,6 +14,7 @@
 #include <linux/spinlock.h>
 #include <linux/circ_buf.h>
 #include <linux/skbuff.h>
+#include <linux/compiler.h>
 
 struct skb_desc {
 	struct sk_buff *skb;
@@ -23,10 +24,10 @@ struct skb_desc {
 struct skb_ring {
 	/* reader lock */
 	spinlock_t rlock;
-	unsigned long tail;
-	unsigned long size;
+	unsigned long tail ____cacheline_aligned_in_smp;
+	unsigned long size ____cacheline_aligned_in_smp;
 	struct skb_desc *descs;
-	unsigned long head;
+	unsigned long head ____cacheline_aligned_in_smp;
         /* writer lock */;
 	spinlock_t wlock;
 };
